nomos-services/carnot-consensus/src/network/mod.rs

```diff
@@ -14,27 +14,41 @@ use nomos_network::NetworkService;
 use overwatch_rs::services::relay::OutboundRelay;
 use overwatch_rs::services::ServiceData;
 
-type BoxedStream<T> = Box<dyn Stream<Item = T> + Send + Sync + Unpin>;
-
-#[async_trait::async_trait]
 pub trait NetworkAdapter {
     type Backend: NetworkBackend + 'static;
-    async fn new(
+    fn new(
         network_relay: OutboundRelay<<NetworkService<Self::Backend> as ServiceData>::Message>,
-    ) -> Self;
-    async fn proposal_chunks_stream(
+    ) -> impl Future<Output = Self> + Send
+    where
+        Self: Sized;
+    fn proposal_chunks_stream(
+        &self,
+        view: View,
+    ) -> impl Future<Output = impl Stream<Item = ProposalMsg> + Send + Sync + Unpin + 'static> + Send;
+    fn broadcast(&self, message: NetworkMessage) -> impl Future<Output = ()> + Send;
+    fn timeout_stream(
+        &self,
+        committee: &Committee,
+        view: View,
+    ) -> impl Future<Output = impl Stream<Item = TimeoutMsg> + Send + Sync + Unpin + 'static> + Send;
+    fn timeout_qc_stream(
```

Comment:
Ufff, seeing this I do not think the code readability is actually improve and all...

Reply:
We can directly use `async` keyword in the trait here, but it will show a warning. We can use `#[allow(...)]` to ignore the warnings. But still, readability is not as good as `BoxStream` and using `async_trait`. For performance, we avoid two `Box` allocations, one for Box stream, and another for Box future.

Reply:
> We can directly use `async` keyword in the trait here, but it will show a warning. We can use `#[allow(...)]` to ignore the warnings. But still, readability is not as good as `BoxStream` and using `async_trait`. For performance, we avoid two `Box` allocations, one for Box stream, and another for Box future.

For now performance is not an issue. I prefer clean code that messy one. I would like to know the opinion of the team on this but we are not in a hurry to change this. I remember reading that they will be improving it (remember something about forcing code to be `send` or not and that there is a macro for it). 
IMO for now we can stall this.

Lets vote on it @bacv @youngjoon-lee @zeegomo 
🟢 Merge 🔴 Keep

Reply:
https://crates.io/crates/trait-variant is the macro that can help, but will introduce an extra useless local trait.

Reply:
> https://crates.io/crates/trait-variant is the macro that can help, but will introduce an extra useless local trait.

Yes, for that is more comfortable to keep the `[async_trait]` for now. This is introducing complexity instead of making it simpler. IMO